contracts/mechanisms/LockDrop.sol

```diff
@@ -1,36 +1,92 @@
 pragma solidity ^0.4.24;
 
 import "openzeppelin-zos/contracts/math/SafeMath.sol";
+import "../token/EdgewareERC20.sol";
 
 /**
  * The LockDrop contract locks up funds of depositors
  * with a verifiable receipt/proof that can be used to
  * redeem tokens on a separate, compatible blockchain.
- * THe tokens with be locked for a specified time length.
+ * The tokens with be locked for a specified time length.
+ * 
+ * The tokens are minted at a rate that grows according
+ * to the chosen locking schedule.
  */
 contract LockDrop {
-    uint public lockDropCapacity;
-    uint public lockDropBeginning;
-    uint public lockDropEnding;
+    uint public capacity;
+    uint public beginning;
+    uint public ending;
+    uint public maxLength;
 
-    mapping (address => uint) lockDropBalances;
+    struct Deposit {
+        uint amount;
+        uint lockEnding;
+    }
+    
+    mapping (address => Deposit[]) deposits;
 
-    constructor(uint length, uint capacity) {
-        lockDropCapacity = capacity;
-        lockDropBeginning = now;
-        lockDropEnding = now * length * 1 day;
+    constructor(uint _lockPeriod, uint _maxLength, uint _capacity) {
```

Comment:
constructor
* Should also specify the specific time lockup schedules for long drop, as well as any additional discounts that are minted for longer lockups
    * i.e.
        * 3 Month cliff
            * 0% Discount
        * 1 years, 6 month cliff
            * 7.5% discount
        * 2 years, 6 month cliff
            * 15% discount
        * 3 years, 6 month cliff
            * 20% discount

This may mean that we have to create an enum/other structure to store discounts

Reply:
The constructor may additionally want to define how many EdgewareERC20 are to be given to public within this LockDrop